File: frontend/src/services/analyticsService.ts
import { apiClient } from '../api/client';
import { ApiResponse } from '../types/api';
import {
  AnalyticsOverviewResponse,
  StatusCountResponse,
  PriorityCountResponse,
  SeverityCountResponse,
  TeamWorkloadResponse,
  SlaAnalyticsResponse,
  KnowledgeAnalyticsResponse,
} from '../types/analytics';

export const analyticsService = {
  async getOverview(from?: string, to?: string): Promise<AnalyticsOverviewResponse> {
    const params: Record<string, string> = {};
    if (from) params.from = from;
    if (to) params.to = to;
    const response = await apiClient.get<ApiResponse<AnalyticsOverviewResponse>>('/analytics/overview', { params });
    return response.data.data;
  },

  async getStatusBreakdown(from?: string, to?: string): Promise<StatusCountResponse[]> {
    const params: Record<string, string> = {};
    if (from) params.from = from;
    if (to) params.to = to;
    const response = await apiClient.get<ApiResponse<StatusCountResponse[]>>('/analytics/incidents/status', { params });
    return response.data.data;
  },

  async getPriorityBreakdown(from?: string, to?: string): Promise<PriorityCountResponse[]> {
    const params: Record<string, string> = {};
    if (from) params.from = from;
    if (to) params.to = to;
    const response = await apiClient.get<ApiResponse<PriorityCountResponse[]>>('/analytics/incidents/priority', { params });
    return response.data.data;
  },

  async getSeverityBreakdown(from?: string, to?: string): Promise<SeverityCountResponse[]> {
    const params: Record<string, string> = {};
    if (from) params.from = from;
    if (to) params.to = to;
    const response = await apiClient.get<ApiResponse<SeverityCountResponse[]>>('/analytics/incidents/severity', { params });
    return response.data.data;
  },

  async getTeamWorkload(): Promise<TeamWorkloadResponse[]> {
    const response = await apiClient.get<ApiResponse<TeamWorkloadResponse[]>>('/analytics/teams/workload');
    return response.data.data;
  },

  async getSlaMetrics(): Promise<SlaAnalyticsResponse> {
    const response = await apiClient.get<ApiResponse<SlaAnalyticsResponse>>('/analytics/sla');
    return response.data.data;
  },

  async getKnowledgeMetrics(): Promise<KnowledgeAnalyticsResponse> {
    const response = await apiClient.get<ApiResponse<KnowledgeAnalyticsResponse>>('/analytics/knowledge');
    return response.data.data;
  },
};

File: backend/src/main/java/com/resolveai/analytics/service/AnalyticsService.java
package com.resolveai.analytics.service;

import com.resolveai.analytics.dto.*;
import com.resolveai.analytics.repository.AnalyticsRepository;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.time.Instant;
import java.util.List;

@Service
@RequiredArgsConstructor
@Slf4j
public class AnalyticsService {

    private final AnalyticsRepository analyticsRepository;

    @Transactional(readOnly = true)
    public AnalyticsOverviewResponse getIncidentOverview(Instant from, Instant to) {
        return analyticsRepository.getIncidentOverview(from, to);
    }

    @Transactional(readOnly = true)
    public List<StatusCountResponse> getStatusBreakdown(Instant from, Instant to) {
        return analyticsRepository.getStatusBreakdown(from, to);
    }

    @Transactional(readOnly = true)
    public List<PriorityCountResponse> getPriorityBreakdown(Instant from, Instant to) {
        return analyticsRepository.getPriorityBreakdown(from, to);
    }

    @Transactional(readOnly = true)
    public List<SeverityCountResponse> getSeverityBreakdown(Instant from, Instant to) {
        return analyticsRepository.getSeverityBreakdown(from, to);
    }

    @Transactional(readOnly = true)
    public List<TeamWorkloadResponse> getTeamWorkload() {
        return analyticsRepository.getTeamWorkload();
    }

    @Transactional(readOnly = true)
    public SlaAnalyticsResponse getSlaMetrics() {
        return analyticsRepository.getSlaMetrics();
    }

    @Transactional(readOnly = true)
    public KnowledgeAnalyticsResponse getKnowledgeMetrics() {
        return analyticsRepository.getKnowledgeMetrics();
    }
}
